src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -381,8 +381,8 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 			theme: this._getXtermTheme(),
 			drawBoldTextInBrightColors: config.drawBoldTextInBrightColors,
 			fontFamily: font.fontFamily,
-			fontWeight: config.fontWeight,
-			fontWeightBold: config.fontWeightBold,
+			fontWeight: config.fontWeight as FontWeight, // TODO: Remove assertions when compatible FontWeignt used in xterm
+			fontWeightBold: config.fontWeightBold as FontWeight,
```

Comment:
@Tyriar I had to add assertion here because Xterm type has only fixed list of string literal types, while there are no problem with passing something like `"350"` which works as expected.
Do you think it's possible to update Xterm's types to support also numbers
`type FontWeight = 'normal' | 'bold' | '100' | ... | '900' | number;` ?

I used `string` instead of `number` in this PR because it kept changes local, but after encountering problem with Xterm it might make sense to use `number`.

Reply:
👍 created https://github.com/xtermjs/xterm.js/issues/3051 to follow that up